### PR TITLE
CB-20674 Cherry-pick changes from master to 2.68

### PR DESCRIPTION
**Pull request by pkedvessy**
See detailed description in the commit message.

### CODE REVIEW

File: orchestrator-salt/src/main/resources/salt/salt/java/init.sls [hunk 1]
```diff
@@ -1,3 +1,17 @@
+/opt/salt/scripts/set_default_java_version.sh:
+  file.managed:
+    - source:
+        - salt://java/scripts/set_default_java_version.sh
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt/salt/java/init.sls:4]: the state file and the script name also should reflect, that this is setting java to the 11 version. 
so it would be better to have a state called java11.sls and also the script name like set_default_java_version_to_11

File: orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    alternatives --set java java-11-openjdk.x86_64
+    ln -sfn /etc/alternatives/java_sdk_11 /usr/lib/jvm/java
+    mkdir -p /etc/alternatives/java_sdk_11/jre/lib/security
+    ln -sfn /etc/alternatives/java_sdk_11/conf/security/java.security /etc/alternatives/java_sdk_11/jre/lib/security/java.security
+    ln -sfn /etc/pki/java/cacerts /etc/alternatives/java_sdk_11/jre/lib/security/cacerts
+
+    #Regenerate the resolv.conf
+    source /etc/dhcp/dhclient-enter-hooks
+    generate_resolv_conf
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:25]: why is this required?

File: orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    echo "Current java version equals to the target version."
+    exit 0;
+  elif [ $CURRENT_JAVA_VERSION -eq 8 ] && [ $TARGET_JAVA_VERSION -eq 11 ]; then
+    echo "Change java version from $CURRENT_JAVA_VERSION to $TARGET_JAVA_VERSION"
+
+    alternatives --set java java-11-openjdk.x86_64
+    ln -sfn /etc/alternatives/java_sdk_11 /usr/lib/jvm/java
+    mkdir -p /etc/alternatives/java_sdk_11/jre/lib/security
+    ln -sfn /etc/alternatives/java_sdk_11/conf/security/java.security /etc/alternatives/java_sdk_11/jre/lib/security/java.security
+    ln -sfn /etc/pki/java/cacerts /etc/alternatives/java_sdk_11/jre/lib/security/cacerts
```
Comment (lacikaaa) [inline orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:21]: is this valid for centos7 and rh8 too?

**Review by lacikaaa — CHANGES_REQUESTED** [review]

File: orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh (hunk 3 above)
Comment (pkedvessy) [reply orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:21]: TBH I don't know, this should be covered when start the support of RHEL8 runtime images.

File: orchestrator-salt/src/main/resources/salt/salt/java/init.sls (hunk 1 above)
Comment (pkedvessy) [reply orchestrator-salt/src/main/resources/salt/salt/java/init.sls:4]: Its not about java 11, it is about changing the default java version to this or that. This script can be extended with magic command for 17 or whatever version later in case of any backward incompatibility.

File: orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh (hunk 3 above)
Comment (pkedvessy) [reply orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:21]: Ticket under the RHEL8 runtime image epic to cover: https://jira.cloudera.com/browse/CB-20820

**Comment by pkedvessy** [comment]
retest this please

File: orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh (hunk 2 above)
Comment (lacikaaa) [reply orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:25]: why is this required to run at this point, when the image already contains the right script?

Comment (pkedvessy) [reply orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:25]: Because the script runs before the orchestration and later as well triggered by the OS. We need the right content when starting CM so we can wait for the OS to run it.

Comment (lacikaaa) [reply orchestrator-salt/src/main/resources/salt/salt/java/scripts/set_default_java_version.sh:25]: so we already have this state: https://github.com/hortonworks/cloudbreak/pull/14247/files#diff-a628a2b611d334912be2f1c73176d08e5ff1123201748b7dbcaf9d207c9a36f5R155
this could be used, preferably with file.replace and with append option, so we could avoid adding this dependency to this script. so there would be only 2 places where this domain thing is touched, instead of 3

**Review by adarshsharma1983 — COMMENTED** [review]
I see that mock-thunderhead changes are not present to add entitlement. Is that intentional?
